src/args.rs

```diff
@@ -111,6 +111,12 @@ pub struct CliArgs {
     #[clap(short = 'u', long = "upload-files")]
     pub file_upload: bool,
 
+    /// Allowed upload directories (together with -u)
+    /// 
+    /// If this is set, uploads are only allowed into the provided directories. 
+    #[clap(long, requires = "file-upload", value_hint = ValueHint::FilePath)]
```

Comment:
If you like, you can choose a short letter for this as well as it'd be annoying to write it out many times otherwise.

Reply:
I'd go for 'A' (as in allowed...), but I am wondering if we might just want to re-purpose `-u`...

Reply:
You mean something like `-u` alone would allow for uploading everywhere and then `-u /tmp/lol` would allow uploading only to that dir? If that's what you're thinking, I appreciate the conciseness but I can't help but wonder whether that wouldn't make `-u` pretty hard to parse (is it even still possible?) and somewhat confusing to users perhaps. Thoughts?

Reply:
`-A` is a fine choice, I think.

Reply:
> You mean something like `-u` alone would allow for uploading everywhere and then `-u /tmp/lol` would allow uploading only to that dir? If that's what you're thinking, I appreciate the conciseness but I can't help but wonder whether that wouldn't make `-u` pretty hard to parse (is it even still possible?) and somewhat confusing to users perhaps. Thoughts?

Exactly, that was my thinking (after seeing so many command line shorts already taken ;)). Not sure about the parsing, though, I'd give it a try...

Reply:
Played around a bit, it looks like it would be relatively easy to parse by using `min_values = 0`:

```rust
    #[clap(short = 'u', long,
     value_hint = ValueHint::FilePath, 
     min_values = 0)]
    pub upload-files: Option<Vec<PathBuf>>,
```

This would yield `None` in case the argument is missing, `Some([])` in case the argument is used without parameters and `Some(['some_dir'])` in case a directory is supplied. So doesn't look to complicated in my view.

@svenstaro what do you think? I would remodel the code to have `-u, --upload-files [<ALLOWED_UPLOAD_DIR>...]` and save the extra argument `-A/--allowed_upload-dir`

Reply:
Does it parse this correctly? `-u --some-other-flag`? Would it try to parse the stuff behind `-u` as a `PathBuf`?

Reply:
Just tried `-u --verbose` and it worked as expected (i.e. verbose output and `allowed_upload_dir == Some([])` (i.e. it would allow uploads everywhere)

Reply:
I've just committed a version that uses `-u` to define allowed directories. @svenstaro, please check it out and let me know what you think (I am also fine to use the previous version with `-A` as a shortcut).

Reply:
I like the `-u` with `min_values`. Makes more sense. Might want to try how it does in windows shell. It should work in bash because we've to use `./-path` when a file starts with `-` (e.g. `touch ./-h`) otherwise it'll be parsed as cli argument.

Reply:
Unfortunately, I cannot try on Windows.

Directories starting with `-` would also need the `./` prefix, which doesn't work with my implementation (yet) - will have a look.